File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/intercept/Intercept.java
package com.adadapted.android.sdk.core.intercept;

import java.util.ArrayList;
import java.util.List;

public class Intercept {
    private static final String SEARCH_ID = "empty";
    private static final long REFRESH_TIME = 300;
    private static final int MIN_MATCH_LENGTH = 3;

    private final String searchId;
    private final long refreshTime;
    private final int minMatchLength;
    private final List<Term> terms;

    public static Intercept empty() {
        return new Intercept(
            SEARCH_ID,
            REFRESH_TIME,
            MIN_MATCH_LENGTH,
            new ArrayList<Term>()
        );
    }

    public Intercept(final String searchId,
                     final long refreshTime,
                     final int minMatchLength,
                     final List<Term> terms) {
        this.searchId = searchId;
        this.refreshTime = refreshTime;
        this.minMatchLength = minMatchLength;
        this.terms = terms;
    }

    public String getSearchId() {
        return searchId;
    }

    public long getRefreshTime() {
        return refreshTime;
    }

    public int getMinMatchLength() {
        return minMatchLength;
    }

    public List<Term> getTerms() {
        return terms;
    }
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/intercept/Intercept.kt
package com.adadapted.android.sdk.core.intercept

import com.google.gson.annotations.SerializedName

class Intercept(
        @SerializedName("search_id")
        val searchId: String = SEARCH_ID,

        @SerializedName("refresh_time")
        val refreshTime: Long = REFRESH_TIME,

        @SerializedName("min_match_length")
        val minMatchLength: Int = MIN_MATCH_LENGTH,

        private val terms: List<Term> = ArrayList()) {

    fun getTerms(): List<Term> {
        return terms.sortedWith(Comparator(Term::compareTo))
    }

    companion object {
        private const val SEARCH_ID = "empty"
        private const val REFRESH_TIME: Long = 300
        private const val MIN_MATCH_LENGTH = 3
    }
}